src/Cis194/Hw/Week1.hs

```diff
@@ -27,7 +34,16 @@ type Peg = String
 type Move = (Peg, Peg)
 
 hanoi :: Integer -> Peg -> Peg -> Peg -> [Move]
-hanoi _ _ _ _ = []
+hanoi 1 a b _ = [(a, b)]
+hanoi 0 _ _ _ = []
+hanoi n a b c = (hanoi (n - 1) a c b) ++ (hanoi 1 a b c) ++ (hanoi (n - 1) c b a)
 
 hanoi4 :: Integer -> Peg -> Peg -> Peg -> Peg -> [Move]
-hanoi4 _ _ _ _ _ = []
+hanoi4 2 a b c _ = [(a, c), (a, b), (c, b)]
+hanoi4 1 a b _ _ = [(a, b)]
+hanoi4 0 _ _ _ _ = []
```

Comment:
All of the `hanoi4` above this point can just delegate to `hanoi` since the outcomes are the same. I spent some time looking at this on mine and trying to see how I could utilize `hanoi` and noticed those outputs are the same.

Reply:
Your comment made me realize that the case `hanoi4 2 a b c _` can be omitted completely (which I just tested)

Reply:
Nice, I didn't think about that.